### DISCUSSION TITLE
How to render field set by mutateFormDataBeforeFill()

### DISCUSSION THREAD

**Discussion by bartclaeys** [filamentphp/filament#5118] [Help] [answered]
Trying to display the author (user) of the record on the Form edit page using `mutateFormDataBeforeFill()` as follows:

```
protected function mutateFormDataBeforeFill(array $data): array
    {
        $data['user'] = User::find($data['created_by']);
        return $data;
    }
```

But not sure how to render this field. Following does not work:

```
Placeholder::make('Created by')
        ->content(function (Book $record) {
            return ($record) ? $record->user : 'Unknown';
    }),
```

**Comment by danharrin** [comment, marked as answer]
You dont need the `mutateFormDataBeforeFill()`, try 

`->content(fn (Book $record) => $record?->createdBy?->name ?? 'Unknown')`

I would suggest adding a `createdBy` BelongsTo relationship to the book

```php
public function createdBy(): BelongsTo
{
    return $this->belongsTo(User::class, 'created_by');
}
```

**Reply by bartclaeys** [reply]
That works, thanks!

**Reply by ArtMin96** [reply]
@bartclaeys Can you mark this as answered so others don't ask the same question again?
